src/main/java/me/juancarloscp52/entropy/server/ServerEventHandler.java

```diff
@@ -123,23 +122,24 @@ public void tick(boolean noNewEvents) {
         eventCountDown--;
     }
 
-    public boolean runEvent(Holder.Reference<EventType<?>> typeReference) {
-        if (typeReference != null) {
-            EventType<?> type = typeReference.value();
-            if (type.doesWorldHaveRequiredFeatures(server.overworld())) {
-                Event event = type.create();
-                Entropy.LOGGER.info("New Event: {} total duration: {}", typeReference.key().location(), event.getDuration());
-                // Start the event and add it to the list.
-                event.init();
-                currentEvents.add(event);
-
-                sendEventToPlayers(event);
-                return true;
-            }
+    public boolean runEvent(Event event) {
+        if (event == null) {
+            Entropy.LOGGER.info("New Event not found");
+            return false;
         }
 
-        Entropy.LOGGER.info("New Event not found");
-        return false;
+        if (!event.getType().doesWorldHaveRequiredFeatures(server.overworld())) {
+            Entropy.LOGGER.info("Tried to run event that requires disabled features");
```

Comment:
Can we somehow show which features are missing? That would help let people turn on the necessary ones.

Reply:
Made something that I believe should work, but nothing requires any features and I don't care enough about a log line to implement a test case.